### PR TITLE
[PLEASE READ THIS PR!!!] RNG REMOVAL PART 1: GENERAL COMBAT

### PR DESCRIPTION
**Pull request by zeskorion**
## About The Pull Request
refactors / reworks the following mechanics to not use rng:
dismemberment
blocking
confusion from being hit
punching
**READ THE DAMN PR AND ALL WILL BE EXPLAINED***
to the best ability of zesko's lazy 3 AM  coedrbrain

**new weapon mechanic**
3/4ths of these changes use a new weapon stat called "attack weight", which represents the weight of a weapon, independent from size or sharpness. most weapons have it at 1, some have it at 2, and a few rare examples have 3

**dismemberment**
dismemberment is calculated using weapon weight, size, sharpness, and force. It's also easier to dismember dead targets if using a sharp weapon. Dismembering targets with blunt objects is possible if the weapon has enough weight

**random confusion, punches, knockdown, etc**
removes all random add effects on normal hits
punching damage is now one flat number

**blocking**
I saved the best for last here

blocking is no longer chance based. Instead, it depends on facing, if the user is walking (walking usually adds a bonus to blocking), if the user is holding the item (most non-shield items must be held for blocking) and if the attack is ranged or not. You also will not block on harm intent.

non-shields don't block projectiles unless they're a double esword or a katana. riot shields are transparent and can't block projectiles that pass through them

there are two kinds of blocking: shield blocking and item blocking. Shields have durability, and break after blocking too many attacks (though energy shields just overheat and must be turned on again), and items cause you to take stamina damage. The calculations behind damage dealt to blocks depends on weight, attacker, sharpness, damage type, etc. 

there are two important numbers for blocking: block Level and block Power. Power is essentially block armor, and determines how hard a block is to break. Block level determines how much a block covers: see shitty diagram below
image

block level 1 is mostly seen on common objects you could expect to parry a blow with, only when walking. Block level 2 is seen on more weaponlike objects when walking, and level 3 is mostly reserved for big shields. Level 4 is seldom seen

also, if you hit someone with your bare hands and get parried by a sharp weapon, it's gonna hurt. 

STAY TUNED FOR RNG HOLOCAUST PART 2: SIMPLEMOBS, EMBEDDING, AND THOSE FUCKING MONKEYS

## Why It's Good For The Game

rng is bad and big gay

## Changelog
:cl:
tweak: no more random confusion when hit in the head
tweak: dismemberment now uses a calculation reliant on weapon size, weight, sharpness, and force to determine when a limb is lost, instead of RNG.
tweak: no more RNG on punches! punches now have fixed damage and no stun.
refactor:  totally changes up block chance. blocking now relies on items being held in the active hand or not, if you are walking or not, and the direction you are facing. Shields now use durability and all other items deal stamina damage to the arm holding them when blocking. Read the pr for more in-depth info.

/:cl:

### CODE REVIEW

**Comment by Coolsurf6** [comment]
Now this is getting spicy

**Comment by CthulhuOnIce** [comment]
Idk about this one chief

**Comment by Cenrus** [comment]
I don't like how attacking the head induces confusion, especially when we still haven't removed wall slams.

**Comment by VictorVow** [comment]
Haven't read all of it yet but yeah either remove wallslams from confusion or make it another status effect that just does the randomised movement.

File: code/game/objects/items.dm [hunk 1]
```diff
@@ -129,14 +129,28 @@ GLOBAL_VAR_INIT(rpg_loot_items, FALSE)
...
+	var/attack_weight = 1
 
 	/// What this thing does when used like a tool. NONE if it isn't a tool. If I give a piece of paper TOOL_WRENCH I can use it to unwrench tables. See _DEFINES/tools.dm
 	var/tool_behaviour = NONE
 	/// The tool speed multiplier of how long it takes to do the tool action.
 	var/toolspeed = 1
 
 	/// The chance that holding this item will block attacks.
-	var/block_chance = 0
+	var/block_level = 0
```
Comment (qwertyquerty) [inline code/game/objects/items.dm:141]: Couldn't 4 of these be bitflags. I'd like to reduce the # of vars on item as much as possible

File: code/game/objects/items.dm [hunk 2]
```diff
@@ -434,13 +448,81 @@ GLOBAL_VAR_INIT(rpg_loot_items, FALSE)
...
+				return 1
+		if(45, 315, -45, -315)
+			if(final_block_level >= 4)
+				playsound(src, block_sound, 50, 1)
+				owner.visible_message("<span class='danger'>[owner] blocks [attack_text] with [src]!</span>")
+				return 1
 	return 0
 
+/obj/item/proc/on_block(mob/living/carbon/human/owner, atom/movable/hitby, attack_text = "the attack", damage = 0, attack_type = MELEE_ATTACK)
+	var/blockhand = 0
```
Comment (qwertyquerty) [inline code/game/objects/items.dm:500]: Chief this proc needs a bit of cleaning

Comment (zeskorion) [reply code/game/objects/items.dm:500]: It's about the cleanest way i could get it to figure out "which hand is this item in"

File: code/game/objects/items.dm (hunk 1 above)
Comment (zeskorion) [reply code/game/objects/items.dm:141]: I have no idea how to use bitflags TBQH

**Comment by CthulhuOnIce** [comment]
Wallslams can be avoided just by walking y'know
@cenrus @Victor239

**Comment by zeskorion** [comment]
The amount of confusion is so small you'll barely notice

**Comment by Cenrus** [comment]
Then why have it in the first place? Walking makes you really easy to click, too.

**Comment by CthulhuOnIce** [comment]
> Then why have it in the first place? Walking makes you really easy to click, too.

dont get confused then

File: code/game/objects/items.dm (hunk 1 above)
Comment (zeskorion) [reply code/game/objects/items.dm:141]: done

**Comment by zeskorion** [comment]
> Then why have it in the first place? Walking makes you really easy to click, too.

you know what, sure, i'll take it out

**Comment by VictorVow** [comment]
> you know what, sure, i'll take it out

Was this changed? If so please update the changelog and OP

**Comment by qwertyquerty** [comment]
vote

**Comment by mehmetipekpatates** [comment]
> You also will not block on harm intent.

What's the point for this? It makes literally no sense.

**Pull request merged**